Libplanet.Headless/Hosting/LibplanetNodeService.cs

```diff
@@ -635,29 +637,53 @@ public override void Dispose()
             Log.Debug("Store disposed.");
         }
 
-        private string ResolvePluginPath(string path) =>
-            Uri.IsWellFormedUriString(path, UriKind.Absolute)
-                ? DownloadPlugin(path).Result
-                : path;
+        private string ResolvePluginPath(string path)
+        {
+            if (Uri.IsWellFormedUriString(path, UriKind.Absolute))
+            {
+                // Run the download on a background thread
+                return Task.Run(() => DownloadPlugin(path, SwarmCancellationToken)).GetAwaiter().GetResult();
```

Comment:
Using `SwarmCancellationToken` here seems invalid. Because `SwarmCancellationToken` is Initialized in `ExecuteAsync -> StartSwarm`. But `ResolvePluginPath` is called in `LibplanetNodeService`'s constructor. 

This may be a sign that you need to separate the downloading logic from LibplanetNodeService.

Reply:
Hmm separating the downloading logic from LibplanetNodeService seems a bit tricky since the ResolvePluginPath(https://github.com/planetarium/NineChronicles.Headless/blob/development/Libplanet.Headless/Hosting/LibplanetNodeService.cs#L126) is called within the LibplanetNodeService constructor. 🤔 

I've removed the wrong use of SwarmCancellationToken entirely for now but do you have any suggestions?

Reply:
For now, I don't have anything else to suggest in this PR: this repository is in need of an overall refactoring. LibplanetNodeService is also code that I wrote when I didn't know .NET DI very well and I think it's actually better to have a generate DI logic that creates an IActionEvaluator and use that IActionEvaluator when generating BlockChain.

Reply:
Ah okok, I understand 🙆‍♂️